Userland/DynamicLoader/main.cpp

```diff
@@ -76,6 +80,30 @@ NAKED void _start(int, char**, char**)
 #endif
 }
 
+static bool open_executable(char const* path, int& main_program_fd)
+{
+    int rc = open(path, O_RDONLY | O_EXEC, 0666);
+    if (rc < 0)
+        return false;
+    int checked_fd = rc;
+    ArmedScopeGuard close_on_failure([checked_fd] {
+        close(checked_fd);
+    });
+    struct stat executable_stat;
+    rc = fstat(checked_fd, &executable_stat);
+    if (rc < 0)
+        return false;
+    if (!S_ISREG(executable_stat.st_mode)) {
+        errno = -EINVAL;
+        if (S_ISDIR(executable_stat.st_mode))
+            errno = -EISDIR;
+        return false;
+    }
+    close_on_failure.disarm();
+    main_program_fd = checked_fd;
+    return true;
+}
```

Comment:
I'm not a fan of this way of handling errors in our C++ code. At the very least, error codes should *not* be negated when assigned to `errno` (have you even tested this?), but I feel like `ErrorOr<void>` would be nicer here.

Reply:
Yes I tested this. I will change this to use `ErrorOr<int>`.